File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;

int main(void){
  ios::sync_with_stdio(0);
  cin.tie(0);
  
  int n;
  cin >> n;

  queue<int> q;
  for(int i=0;i<n;i++){
    string command;
    cin >> command;
    if(command == "push"){
      int num;
      cin >> num;
      q.push(num);
    } else if (command == "pop"){
      if(q.empty()){
        cout << -1 << '\n';
      } else {
        cout << q.front() << '\n';
        q.pop();
      }
    } else if (command == "front"){
      if(q.empty()){
        cout << -1 << '\n';
      } else {
        cout << q.front() << '\n';
      }
    } else if (command == "back"){
      if(q.empty()){
        cout << -1 << '\n';
      } else {
        cout << q.back() << '\n';
      }
    } else if (command == "size"){
      cout << q.size() << '\n';
    } else if (command == "empty"){
      if(q.empty()){
        cout << 1 << '\n';
      } else {
        cout << 0 << '\n';
      }
    }
  }

  return 0;
}

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = null;
    Queue<Integer> Q = new LinkedList<>();
    int n = Integer.parseInt(br.readLine());

    while (n-- > 0) {
      st = new StringTokenizer(br.readLine());
      String cmd = st.nextToken();
      switch (cmd) {
        case "push":
          int val = Integer.parseInt(st.nextToken());
          Q.add(val);
          break;
        case "pop":
          if (Q.isEmpty()) {
            System.out.println(-1);
          } else {
            System.out.println(Q.poll());
          }
          break;
        case "size":
          System.out.println(Q.size());
          break;
        case "empty":
          System.out.println(Q.isEmpty() ? 1 : 0);
          break;
        case "front":
          if (Q.isEmpty()) {
            System.out.println(-1);
          } else {
            System.out.println(Q.peek());
          }
          break;
        case "back":
          if (Q.isEmpty()) {
            System.out.println(-1);
          } else {
            System.out.println(((LinkedList<Integer>) Q).getLast());
          }
          break;
      }
    }
  }
}